lib/nodejs/lobby.js

```diff
@@ -44,6 +44,24 @@ app.get( "/", function( req, res ) {
   } );
 } );
 
+app.get( "/map-list", function( req, res ) {
+  var mapDir = "documents/ITDG/maps";
+
+  // Ensure that the map directory exists, then get the list of files in the map directory
+  mkdirp( mapDir ).then( function( err ) {
```

Comment:
The promise doesn't pass an error to the callback (well, [mostly](http://exploringjs.com/es6/ch_promises.html#_mistake-using-then-for-error-handling)). Errors will pass through the chain and can be handled in a `catch` clause at the end.